stellargraph/datasets/dataset_loader.py

```diff
@@ -0,0 +1,218 @@
+
+    @property
+    def data_directory(self) -> str:
+        """str: The full path of the directory containing the data content files for this dataset."""
+        if self.data_subdirectory_name is None:
+            return self.base_directory
+        else:
+            return os.path.join(self.base_directory, self.data_subdirectory_name)
+
+    def _create_base_directory(self) -> None:
+        data_dir = self.base_directory
+        if not os.path.exists(data_dir):
+            os.makedirs(data_dir)
+
+    @staticmethod
+    def _all_datasets_directory() -> str:
+        """Return the path of the base directory which contains subdirectories for each dataset."""
+        return os.path.expanduser(
+            os.getenv(
+                "STELLARGRAPH_DATASETS_PATH", os.path.join("~", "stellargraph-datasets")
+            )
+        )
+
+    def _resolve_path(self, filename: str) -> str:
+        """Convert dataset relative file names to their full path on filesystem"""
+        return os.path.join(self.base_directory, filename)
+
+    def _is_downloaded(self) -> bool:
+        """Returns true if the expected files for the dataset are present"""
+        return all(
+            os.path.isfile(self._resolve_path(file)) for file in self.expected_files
+        )
+
+    def _verify_files_downloaded(self) -> None:
+        """
+        Raises:
+            FileNotFoundError: If any files within dataset are missing.
+        """
+        missing_files = ", ".join(
+            [
+                file
+                for file in self.expected_files
+                if not os.path.isfile(self._resolve_path(file))
+            ]
+        )
+        if missing_files:
+            raise FileNotFoundError(
+                f"{self.name} dataset failed to download file(s): {missing_files} to {self.data_directory}"
+            )
+
+    def _delete_existing_files(self) -> None:
+        """ Delete the files for this dataset if they already exist """
+        for file in self.expected_files:
+            try:
+                os.remove(self._resolve_path(file))
+            except OSError:
+                pass
+
+    def download(self, ignore_cache: Optional[bool] = False) -> None:
+        """
+        Download the dataset (if not already downloaded)
+
+        Args:
+            ignore_cache bool, optional (default=False): Ignore a cached dataset and force a re-download.
+
+        Raises:
+            FileNotFoundError: If the dataset is not successfully downloaded.
+        """
+        if ignore_cache:
+            self._delete_existing_files()  # remove any existing dataset files to ensure we re-download
+
+        if ignore_cache or not self._is_downloaded():
+            log.info(
+                "%s dataset downloading to %s from %s",
+                self.name,
+                self.base_directory,
+                self.url,
+            )
+            filename, _ = urlretrieve(
```

Comment:
It looks like the comment at the end of this line is making black do this horrible line breaking; maybe:

```suggestion
            # this will download to a temporary location
            filename, _ = urlretrieve(self.url)
```

or potentially even remove the comment, e.g.:

```suggestion
            temporary_filename, _ = urlretrieve(self.url)
```